onnxruntime/core/providers/qnn/builder/opbuilder/conv_op_builder.cc

```diff
@@ -59,16 +59,27 @@ Status ConvOpBuilder::IsOpSupported(QnnModelWrapper& qnn_model_wrapper,
                                     const NodeUnit& node_unit,
                                     const logging::Logger& logger,
                                     bool is_quantized_model) const {
-  ORT_UNUSED_PARAMETER(qnn_model_wrapper);
-  ORT_UNUSED_PARAMETER(node_unit);
-  ORT_UNUSED_PARAMETER(logger);
-  ORT_UNUSED_PARAMETER(is_quantized_model);
-  auto input_0 = node_unit.Inputs()[0];
+  if (node_unit.Domain() == kMSInternalNHWCDomain) {  // Use QNN validation API if layout is NHWC.
+    return AddToModelBuilder(qnn_model_wrapper, node_unit, logger, is_quantized_model, true);
+  }
+
+  const auto& input_0 = node_unit.Inputs()[0];
   std::vector<uint32_t> input_shape;
   ORT_RETURN_IF_NOT(qnn_model_wrapper.GetOnnxShape(input_0.node_arg, input_shape), "Cannot get shape");
   if (input_shape.size() != 4) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "QNN Conv only support 2D!");
   }
+
+  ONNX_NAMESPACE::DataType input_data_type = input_0.node_arg.Type();
+  ORT_RETURN_IF(!is_quantized_model && input_data_type != ONNX_NAMESPACE::Utils::DataTypeUtils::ToType("float"),
+                "QNN EP: Data type ", input_data_type->c_str(),
+                " is not supported for Conv operator in CPU backend.");
+
+  NodeAttrHelper node_helper(node_unit);
+  auto auto_pad = node_helper.Get("auto_pad", std::string("NOTSET"));
+  ORT_RETURN_IF(auto_pad != "NOTSET" && auto_pad != "SAME_LOWER" && auto_pad != "SAME_UPPER",
+                "QNN Conv operators do not support 'auto_pad' value: ", auto_pad.c_str());
+
```

Comment:
Fixes test errors that snuck in while Linux QNN Pipeline did not run unit tests. See https://github.com/microsoft/onnxruntime/pull/15497